qt/scientific_interfaces/Indirect/test/IndirectFitAnalysisTabTest.h

```diff
@@ -0,0 +1,56 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTIDQT_INDIRECTFITANALYSISTABTEST_H_
+#define MANTIDQT_INDIRECTFITANALYSISTABTEST_H_
+
+#include "IndirectFitAnalysisTab.h"
+#include "MantidAPI/FunctionFactory.h"
+#include <cxxtest/TestSuite.h>
+
+class IndirectFitAnalysisTabTest : public CxxTest::TestSuite {
+public:
+  /// Needed to make sure everything is initialized
+  IndirectFitAnalysisTabTest() {}
+
+  static IndirectFitAnalysisTabTest *createSuite() {
+    return new IndirectFitAnalysisTabTest();
+  }
+
+  static void destroySuite(IndirectFitAnalysisTabTest *suite) { delete suite; }
+
+  void test_that_single_function_correctly_identified() {
+    std::string functionName = "ExpDecay";
+    auto fitFunction =
+        Mantid::API::FunctionFactory::Instance().createFunction(functionName);
+    auto occurances =
+        getNumberOfSpecificFunctionContained(functionName, fitFunction);
+    TS_ASSERT_EQUALS(occurances, 1);
+  }
+
+  void test_that_single_layer_composite_function_handled_correctly() {
+    std::string functionName = "name=ExpDecay;name=StretchExp";
+    auto fitFunction =
+        Mantid::API::FunctionFactory::Instance().createInitialized(
+            functionName);
+    auto occurances =
+        getNumberOfSpecificFunctionContained("ExpDecay", fitFunction);
+    TS_ASSERT_EQUALS(occurances, 1);
+  }
+
+  void test_that_multi_layer_composite_function_handled_correctly() {
+    std::string functionName = "name=ExpDecay;name=ExpDecay;(composite="
+                               "ProductFunction,NumDeriv=false;name=ExpDecay;"
+                               "name=ExpDecay)";
+    auto fitFunction =
+        Mantid::API::FunctionFactory::Instance().createInitialized(
+            functionName);
+    auto occurances =
+        getNumberOfSpecificFunctionContained("ExpDecay", fitFunction);
+    TS_ASSERT_EQUALS(occurances, 4);
+  }
+};
+#endif
```

Comment:
```suggestion
#endif

```

Reply:
getNumberOfSpecificFunctionContained is a generic function which just returns the number of any function passed in. I have added a test that it can recognize stretch functions as well. Testing that the 1S is returned correctly would involve adding tests to IqtFit.cpp more specifically the fitTypeString function. I agree completely that this should be tested as well but as that class currently has no tests at all this may take some setup time to do properly and I quite want to get this PR in promptly for the release.

Reply:
That is reasonable given the time constraints.